main/options/options.h

```diff
@@ -170,14 +169,44 @@ struct Options {
     bool enableCounters = false;
     std::string errorUrlBase = "https://srb.help/";
     bool ruby3KeywordArgs = false;
-    bool typedSuper = true;
     std::vector<std::string> suppressPayloadSuperclassRedefinitionFor;
     std::set<int> isolateErrorCode;
     std::set<int> suppressErrorCode;
     bool noErrorSections = false;
     /** Prefix to remove from all printed paths. */
     std::string pathPrefix;
 
+    // Options which affect the contents of the `--cache-dir`.
+    // If these options change, the cache needs to be invalidated.
+    struct CacheSensitiveOptions {
```

Comment:
Could this type be used in GlobalState for the equivalent field, or the other way around? It would be nice to ensure that the two fields be kept in sync, and having them be the same seems like the easiest way to make that happen.

Reply:
2 seems reasonable to me. The thing I was hoping for was that it would be easy to ensure that the two structures don't get out of sync, but we probably won't update this often enough to think to hard about it.

Reply:
https://github.com/sorbet/sorbet/pull/8692